src/coreclr/interpreter/compiler.cpp

```diff
@@ -1128,51 +1185,33 @@ class InterpGcSlotAllocator
             startOffset, endOffset
         );
 
-
-        ConservativeRange* pExistingRange = FindAndRemoveOverlappingOrAdjacentRange(offsetBytes, startOffset, endOffset);
-        if (pExistingRange != nullptr)
-        {
-            INTERP_DUMP("Merging with existing range [%u - %u]\n", pExistingRange->startOffset, pExistingRange->endOffset);
-            pExistingRange->MergeWith(startOffset, endOffset);
-            ConservativeRange* pOtherExistingRange = FindAndRemoveOverlappingOrAdjacentRange(offsetBytes, pExistingRange->startOffset, pExistingRange->endOffset);
-            while (pOtherExistingRange != nullptr)
-            {
-                INTERP_DUMP("Merging with existing range [%u - %u]\n", pOtherExistingRange->startOffset, pOtherExistingRange->endOffset);
-                pExistingRange->MergeWith(pOtherExistingRange->startOffset, pOtherExistingRange->endOffset);
-                pOtherExistingRange = FindAndRemoveOverlappingOrAdjacentRange(offsetBytes, pExistingRange->startOffset, pExistingRange->endOffset);
-            };
-            InsertConservativeRange(offsetBytes, pExistingRange);
-        }
-        else
-        {
-            ConservativeRange* pNewRange = new (m_compiler) ConservativeRange(startOffset, endOffset);
-            InsertConservativeRange(offsetBytes, pNewRange);
-        }
+        uint32_t slotIndex = offsetBytes / sizeof(void *);
+        m_conservativeRanges[slotIndex].InsertRange(m_compiler, startOffset, endOffset);
```

Comment:
Should we be bounds-checking slotIndex? Or are we sure it will always be in range by construction?